### PR TITLE
Support SPDX License List 3.23

### PR DESCRIPTION
**Pull request by AyanSinhaMahapatra**
Reference: https://github.com/nexB/scancode-licensedb/issues/49

### Tasks

* [x] Reviewed [contribution guidelines](https://github.com/nexB/scancode-toolkit/blob/develop/CONTRIBUTING.rst)
* [x] PR is descriptively titled 📑 and links the original issue above 🔗
* [x] Tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR
  Run [tests](https://scancode-toolkit.readthedocs.io/en/latest/contribute/contrib_dev.html#running-tests) locally to check for errors. 
* [x] Commits are in uniquely-named feature branch and has no merge conflicts 📁
* [ ] Updated documentation pages (if applicable)
* [ ] Updated CHANGELOG.rst (if applicable)

### CODE REVIEW

**Review by pombredanne — APPROVED** [review]
LGTM! Thanks

File: src/licensedcode/data/rules/fsfap-no-warranty-disclaimer_1.RULE [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+license_expression: fsfap-no-warranty-disclaimer
+is_license_text: yes
+minimum_coverage: 99
+ignorable_copyrights:
+    - Copyright (c) 2008 Micah J. Cowan
+ignorable_holders:
+    - Micah J. Cowan
+---
+
+Copyright (C) 2008 Micah J. Cowan
```
Comment (pombredanne) [inline src/licensedcode/data/rules/fsfap-no-warranty-disclaimer_1.RULE]: Please do not add a rule with a copyright here. It adds nothing new... copyright will be detected separately ... here this is not a copyright for the license but the code.

File: src/licensedcode/data/rules/gtkbook_1.RULE [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+license_expression: gtkbook
+is_license_text: yes
+minimum_coverage: 98
+ignorable_copyrights:
+    - Copyright 2005 Syd Logan
+ignorable_holders:
+    - Syd Logan
+---
+
+Copyright 2005 Syd Logan, All Rights Reserved
```
Comment (pombredanne) [inline src/licensedcode/data/rules/gtkbook_1.RULE]: Same as above... no copyright please

File: src/licensedcode/data/rules/dec-3-clause_1.RULE [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+license_expression: dec-3-clause
+is_license_text: yes
+minimum_coverage: 90
+ignorable_copyrights:
+    - Copyright 1997 Digital Equipment Corporation
+ignorable_holders:
+    - Digital Equipment Corporation
+---
+
+Copyright 1997 Digital Equipment Corporation.
```
Comment (pombredanne) [inline src/licensedcode/data/rules/dec-3-clause_1.RULE]: No copyrights please

File: src/licensedcode/data/rules/caldera-no-preamble_1.RULE [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+license_expression: caldera-no-preamble
+is_license_text: yes
+minimum_coverage: 98
+ignorable_copyrights:
+    - Copyright (c) Caldera International Inc. 2001-2002
+ignorable_holders:
+    - Caldera International Inc.
+---
+
+Copyright(C) Caldera International Inc.  2001-2002.  All rights reserved.
```
Comment (pombredanne) [inline src/licensedcode/data/rules/caldera-no-preamble_1.RULE]: No copyright here too

File: src/licensedcode/data/rules/other-permissive_242.RULE [hunk 5]
```diff
@@ -1,8 +1,11 @@
 ---
 license_expression: other-permissive
 is_license_text: yes
-relevance: 100
-notes: Seen in mailprio
+is_deprecated: yes
```
Comment (pombredanne) [inline src/licensedcode/data/rules/other-permissive_242.RULE:4]: Add also https://github.com/nexB/scancode-toolkit/blob/5a39c2165282731d86fcc7f99cd0677b91a6e389/src/licensedcode/data/rules/other-permissive_241.RULE and https://github.com/nexB/scancode-toolkit/blob/5a39c2165282731d86fcc7f99cd0677b91a6e389/src/licensedcode/data/rules/other-permissive_243.RULE to point to the new mailprio license

File: src/licensedcode/data/rules/pnmstitch_1.RULE [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+license_expression: pnmstitch
+is_license_text: yes
+minimum_coverage: 95
+ignorable_copyrights:
+    - Copyright (c) 2002 Mark Salyzyn
+ignorable_holders:
+    - Mark Salyzyn
+---
+
+Copyright (c) 2002 Mark Salyzyn
```
Comment (pombredanne) [inline src/licensedcode/data/rules/pnmstitch_1.RULE]: No copyright there

File: src/licensedcode/data/rules/softsurfer_2.RULE [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+minimum_coverage: 95
+ignorable_copyrights:
+    - Copyright 2001, softSurfer (www.softsurfer.com)
+ignorable_holders:
+    - softSurfer
+ignorable_urls:
+    - http://www.softsurfer.com/
+---
+
+Copyright 2001, softSurfer (www.softsurfer.com)
```
Comment (pombredanne) [inline src/licensedcode/data/rules/softsurfer_2.RULE]: No copyright

File: src/licensedcode/data/rules/sun-ppp_1.RULE [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+license_expression: sun-ppp
+is_license_text: yes
+minimum_coverage: 99
+ignorable_copyrights:
+    - Copyright (c) 2001 by Sun Microsystems, Inc.
+ignorable_holders:
+    - Sun Microsystems, Inc.
+---
+
+Copyright (c) 2001 by Sun Microsystems, Inc.
```
Comment (pombredanne) [inline src/licensedcode/data/rules/sun-ppp_1.RULE]: No copyright

File: src/licensedcode/data/rules/umich-merit_1.RULE [hunk 9]
```diff
@@ -0,0 +1,25 @@
+---
+license_expression: umich-merit
+is_license_text: yes
+minimum_coverage: 95
+---
+
+[C] The Regents of the University of Michigan and Merit Network, Inc. 1992,
```
Comment (pombredanne) [inline src/licensedcode/data/rules/umich-merit_1.RULE:7]: See https://github.com/nexB/scancode-toolkit/issues/3659 for reference ./.. so we can keep this copyright for now.

File: src/licensedcode/data/rules/xkeyboard-config-zinoviev_1.RULE [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+minimum_coverage: 95
+ignorable_copyrights:
+    - Copyright (c) 1999, 2000 by Anton Zinoviev <anton@lml.bas.bg>
+ignorable_holders:
+    - Anton Zinoviev
+ignorable_emails:
+    - anton@lml.bas.bg
+---
+
+Copyright (C) 1999, 2000 by Anton Zinoviev <anton@lml.bas.bg>
```
Comment (pombredanne) [inline src/licensedcode/data/rules/xkeyboard-config-zinoviev_1.RULE]: No copyright

**Review by pombredanne — APPROVED** [review]
Here are a few nits on copyrights... and good to merge otherwise!
Thanks

File: src/licensedcode/data/licenses/drl-1.1.LICENSE [hunk 11]
```diff
@@ -5,44 +5,27 @@ name: Detection Rule License (DRL) 1.1
...
-
-THE RULES ARE PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR 
-IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, 
-FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE 
-AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER 
-LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, 
-OUT OF OR IN CONNECTION WITH THE RULES OR THE USE OR OTHER DEALINGS IN THE 
-RULES.
\ No newline at end of file
+Permission is hereby granted, free of charge, to any person obtaining a copy of this rule set and associated documentation files (the "Rules"), to deal in the Rules without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Rules, and to permit persons to whom the Rules are furnished to do so, subject to the following conditions:
```
Comment (pombredanne) [inline src/licensedcode/data/licenses/drl-1.1.LICENSE]: Why did you replace our text which is better and properly formatted?
You should IMHO keep our text and add the SPDX text as a detection rule instead.

File: src/licensedcode/data/licenses/hdparm.LICENSE [hunk 12]
```diff
@@ -4,16 +4,22 @@ short_name: HDPARM License
...
+other_urls:
+    - https://github.com/Distrotech/hdparm/blob/4517550db29a91420fb2b020349523b1b4512df2/LICENSE.TXT
 minimum_coverage: 90
 ignorable_emails:
     - mlord@pobox.com
 ---
 
-You may freely use, modify, and redistribute the hdparm program, as either
-binary or source, or both.
+BSD-Style Open Source License:
```
Comment (pombredanne) [inline src/licensedcode/data/licenses/hdparm.LICENSE]: Please keep our original text and add the SPDX text as a detection rule

File: src/licensedcode/data/licenses/paul-mackerras.LICENSE [hunk 13]
```diff
@@ -27,7 +29,7 @@ are met:
 3. Redistributions of any form whatsoever must retain the following
    acknowledgment:
    "This product includes software developed by Paul Mackerras
-    <paulus@samba.org>".
+    <paulus@ozlabs.org>".
```
Comment (pombredanne) [inline src/licensedcode/data/licenses/paul-mackerras.LICENSE]: Please add our original text with samba.org as a rule too

File: src/licensedcode/data/licenses/softsurfer.LICENSE [hunk 14]
```diff
@@ -5,13 +5,17 @@ name: softSurfer License
...
+other_spdx_license_keys:
+    - LicenseRef-scancode-softsurfer
 other_urls:
     - http://geomalgorithms.com/index.html
+    - https://fedoraproject.org/wiki/Licensing/softSurfer
+    - https://github.com/mm2/Little-CMS/blob/master/src/cmssm.c#L207
 ---
 
-This code may be freely used and modified for any purpose
+This code may be freely used and modified for any purpose
```
Comment (pombredanne) [inline src/licensedcode/data/licenses/softsurfer.LICENSE]: Please keep our text as-is and add a rule for the SPDX license text

File: src/licensedcode/data/licenses/tgppl-1.0.LICENSE [hunk 15]
```diff
@@ -5,202 +5,199 @@ name: Transitive Grace Period Public Licence 1.0
...
     - Copyright (c) 2007 Zooko
 ignorable_holders:
     - Zooko
 ---
 
+=======================================================
 Transitive Grace Period Public Licence ("TGPPL") v. 1.0
+=======================================================
 
-This Transitive Grace Period Public Licence (the "License") applies to
```
Comment (pombredanne) [inline src/licensedcode/data/licenses/tgppl-1.0.LICENSE]: Please our text as-is and add add a new rule with the SPDX text

**Review by pombredanne — CHANGES_REQUESTED** [review]
Please our texts as-is and add add new rules with the SPDX texts rather than modifying our texts.

**Comment by AyanSinhaMahapatra** [comment]
@pombredanne thanks for the comments, updated them to preserve the original texts.

**Review by DennisClark — APPROVED** [review]
@AyanSinhaMahapatra   I tried really hard to find problems, but did not succeed!  This looks ready to go, thanks.

**Comment by AyanSinhaMahapatra** [comment]
Thanks @DennisClark @pombredanne , merging!

**Pull request merged**